sia/hostdb/hostannouncement.go

```diff
@@ -0,0 +1,83 @@
+package hostdb
+
+import (
+	"github.com/NebulousLabs/Sia/consensus"
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/network"
+)
+
+// A HostAnnouncement is a struct that can appear in the arbitrary data field.
+// It is preceded by 8 bytes that decode to the integer 1.
+type HostAnnouncement struct {
+	IPAddress          network.Address
+	TotalStorage       int64 // Can go negative.
+	MinFilesize        uint64
+	MaxFilesize        uint64
+	MinDuration        consensus.BlockHeight
+	MaxDuration        consensus.BlockHeight
+	MinChallengeWindow consensus.BlockHeight
+	MaxChallengeWindow consensus.BlockHeight
+	MinTolerance       uint64
+	Price              consensus.Currency
+	Burn               consensus.Currency
+	CoinAddress        consensus.CoinAddress // Host may want to give different addresses to each client.
+
+	SpendConditions consensus.SpendConditions
+	FreezeIndex     uint64 // The index of the output that froze coins.
+}
+
+func findHostAnnouncements(height consensus.BlockHeight, b consensus.Block) (entries []HostEntry, err error) {
+	for _, t := range b.Transactions {
+		// Check the arbitrary data of the transaction to fill out the host database.
+		if len(t.ArbitraryData) == 0 {
+			continue
+		}
+		if len(t.ArbitraryData[0]) < 8 {
+			continue
+		}
```

Comment:
code like this makes me question the decision to make ArbitraryData a `[]string`. What's the payoff again?